### DISCUSSION TITLE
How to extract room count and numbers from Xiaomi/Vac without map card?

### DISCUSSION THREAD

**Discussion by fokcuk** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#470] [Q&A] [answered]
Hi
Can you please hint on how to extract the available room count for the vacuum? My vac resets maps often and its a pain to rename all of my automations with the new room numbers. I just want it to get a total room count and when we are not at home HA will send it to a random room to clean

**Comment by PiotrMachowski** [comment, marked as answer]
I have provided you a link in original question. You have to add `rooms` or `room_numbers` in config of map extractor

https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/discussions/317

**Reply by fokcuk** [reply]
I understand all that. And I get it through your integration. I am just asking how you managed to extract that info from API =)

**Reply by PiotrMachowski** [reply]
It isn't retrieved directly from API, it is retrieved from the map data during parsing

**Reply by fokcuk** [reply]
Ah.. Ok. Thank you. Will use map attributes then =)
